### ISSUE TITLE
A tutorial for user to train a customed KG and extract the generated embeddings

### ISSUE THREAD

**Issue by chanlevan** [Accenture/AmpliGraph#67]
**Background and Context**
Users are interested in how to train their KG, the generated embeddings and the visualisation of the embedding.  
**Description**
We should have a concrete code example to show the users how to do train, extract and visualisation the embeddings. Moreover, there should be a visualisation API integrated in the library so that the users can use it.

**Comment by NicholasMcCarthy** [comment]
I guess the simplest way to incorporate this is by having a tutorials directory?

**Comment by chanlevan** [comment]
It sounds ok with me. @lukostaz how about your opinion?

**Comment by tabacof** [comment]
I have added to branch [feature/67](https://github.com/Accenture/AmpliGraph/tree/feature/67) a new tutorial, that 1. creates a triples dataset from tabular data of international football matches 2. trains KGE 3. evaluate them 4. cluster them on the embedding space and compare with natural continental clusters 5. use the embeddings in a downstream classification task to predict match winners (with out of time test set, which makes it particularly hard). @sumitpai has already reviewed this tutorial and it is updated with his feedback.

I have also updated the Basics tutorial to make it compatible with AmpliGraph's develop version, fixed some loose ends, added a few comments here and there, and made section 7 (Predicting New Links) more visually appealing by displaying the results as a Pandas dataframe. This updated notebook has not been reviewed. Volunteers?

**Comment by lukostaz** [comment]
@tabacof leave the review to me.

**Comment by lukostaz** [comment]
Some comments:
- Introduction: 
	- Emphasize in text that the ultimate goal and main focus of the tutorial is the link-based cluster analysis part and the classification. Explain why they are hard tasks and why it KG embeddings are useful. - The KG creation is a required step, but it is after all a necessary step to the ultimate goal.
- Dataset:
	- explain that the dataset comes in tabular form, and that you need to convert it into a KG
	- Is train/test original split necessary for this tutorial or does it only add complexity?
	- Add code snippet to automatically download the tabular dataset from kaggle
- KG creation:
	- Add a visual depiction of the KG schema (just like you did for the maracanazzo, but with classes instead of actual instances). This should follow the first sentences after the title, so that users immediately know at a glance how the graph will look like.
	- stress that cell6-9 is pre-processing required to convert the tabular dataset into a graph. You may want to include the link to the final KG explicitly in the text, so that is anybody wants to start from there they don't have to go through pre-processing again. We can upload the KG on the S3 bucket. Just let me know.
	- maracanazzo: rename `homeTeam`-->`isHomeTeamIn`, `awayTeam`-->`isAwayTeamIn` to make it semantically meaningful or reverse the direction of arrows (the latter requires changes in the data, though)
	- maracanazzo: in the picture "Match" should be the actual instance, "Match3129". homeScores and awayScores are not in the pic. See above.
	- Cell 8: What happened to `isNeutral`, `atYear`, `homeScores`, `awayScores`? Since they are numerics literals, you should say if you filter them out before generating the embeddings or if they are still part of the graph.
- Clustering and embedding visualization
	- The first pic shows the ground truth, right? Can you please clarify a bit and better highlight the 
	- The two cluster pics should have the same labels, if possible.

**Comment by tabacof** [comment]
>Emphasize in text that the ultimate goal and main focus of the tutorial is the link-based cluster analysis part and the classification. Explain why they are hard tasks and why it KG embeddings are useful. - The KG creation is a required step, but it is after all a necessary step to the ultimate goal.

Done.

>explain that the dataset comes in tabular form, and that you need to convert it into a KG

Done.

>Is train/test original split necessary for this tutorial or does it only add complexity?

Yes, otherwise there will be a train-test leakage via the embeddings, which would make the classification example totally unreliable. Also, the temporal split is more realistic for this kind of predictive problem, and it demonstrates that embeddings as features can actually generalise out of time, a lot more useful than just overfitting to the past.

>Add code snippet to automatically download the tabular dataset from kaggle

That is not possible, unless we re-upload the dataset to our own server. Kaggle requires to be logged in to download this dataset. I feel queasy about re-uploading the same dataset somewhere else, even if it is public domain.

>Add a visual depiction of the KG schema (just like you did for the maracanazzo, but with classes instead of actual instances). This should follow the first sentences after the title, so that users immediately know at a glance how the graph will look like.

I don't think doing a more abstract visual presentation will be helpful as people cling to the concrete rather than the abstract. Our own docs present a concrete example in the landing page rather an abstract one. I already spent quite some time doing the visualisation for this tutorial, if possible I'd like to move on.

>stress that cell6-9 is pre-processing required to convert the tabular dataset into a graph. You may want to include the link to the final KG explicitly in the text, so that is anybody wants to start from there they don't have to go through pre-processing again. We can upload the KG on the S3 bucket. Just let me know.

Added more explanations of the rationale of going from tabular to KGE. The KGE generation is quite fast itself, doesn't even take a minute. The only advantage of providing the triples dataset is to remove the necessity of the user having to download the original Kaggle dataset, so that is something we should consider.

>maracanazzo: rename homeTeam-->isHomeTeamIn, awayTeam-->isAwayTeamIn to make it semantically meaningful or reverse the direction of arrows (the latter requires changes in the data, though)

Done.

>maracanazzo: in the picture "Match" should be the actual instance, "Match3129". homeScores and awayScores are not in the pic. See above.

Done. Added explanation that not all relations are present in the visualisation, but they can be seen in the dataframe below.

>Cell 8: What happened to isNeutral, atYear, homeScores, awayScores? Since they are numerics literals, you should say if you filter them out before generating the embeddings or if they are still part of the graph.

Added explanation of how literals are treated as entities in this case.

>The first pic shows the ground truth, right? Can you please clarify a bit and better highlight the

Highlight the?

>The two cluster pics should have the same labels, if possible.

Done.
